### ISSUE TITLE
nre doc errors?

### ISSUE THREAD

**Issue by mark-summerfield** [nim-lang/Nim#5542]
The nre module's docs includes these examples:
```
    "abc".match(re"(\w)").captures[0] == "a"
    "abc".match(re"(?<letter>\w)").captures["letter"] == "a"
    "abc".match(re"(\w)\w").captures[-1] == "ab"
    "abc".match(re"(\w)").captureBounds[0] == 0 .. 0
    "abc".match(re"").captureBounds[-1] == 0 .. -1
    "abc".match(re"abc").captureBounds[-1] == 0 .. 2
```
But my reading of the docs (and trying it out) shows that match() returns an Option[nre.RegexMatch] so all these examples need a get() call to work.

Also, for 1.0 I really hope you choose re or nre rather than have both. Since they both wrap pcre presumably they can both deliver the same functionality, so my suggestion would be to stick with the one you find easiest to maintain and extend.

**Comment by Araq** [comment]
IMO we should move nre to its own nimble package and clean up ``re.nim``.

**Comment by narimiran** [comment]
Fixed by #9053. Should be closed.

**Comment by flaviut** [comment]
Agreed. First paragraph dupes #9053 (which is fixed), second paragraph dupes #7278 (which is still open).

Thank you @narimiran for the tip!
